build.sh

```diff
@@ -10,6 +10,8 @@ if [ -z "${HOME:-}" ]; then
     mkdir "$HOME"
 fi
 
+source "$SCRIPT_ROOT/check-submodules.sh"
```

Comment:
Can you include an environment variable check or something so these can be disabled? Running without all submodules and/or with local changes is a typical dev flow for me, and I don't want to have to hit "n" interactively to get my build going.

I'm also curious how long these commands take, in case that's also something to consider.

Reply:
Added `SOURCE_BUILD_SKIP_SUBMODULE_CHECK`, any value other than `0` or `false` will turn it off.  This is taking about 17 seconds on my Windows machine and 2 seconds on Linux.

Reply:
That's without any fixes necessary - submodules initialized, up-to-date, and clean.